File: java/000/TreeNode.java

public class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;

    TreeNode(int x) {
        val = x;
    }

    public static TreeNode getTreeNode(int... is) {
        TreeNode[] tns = new TreeNode[is.length];
        for (int i = 0; i < tns.length; i++)
            tns[i] = new TreeNode(is[i]);
        for (int i = 0; i < tns.length; i++) {
            int left = 2 * i + 1, right = left + 1;
            if (left >= tns.length) break;
            tns[i].left = tns[left];
            if (right >= tns.length) break;
            tns[i].right = tns[right];
        }
        return tns[0];
    }

    public static void printTreeNode(TreeNode... tns) {
        for (TreeNode tn : tns) {
            print(tn);
            System.out.println();
        }
    }

    private static void print(TreeNode tn) {
        if (tn == null) return;
        System.out.print('[');
        print(tn.left);
        System.out.print(tn.val);
        print(tn.right);
        System.out.print(']');
    }
}

File: swift/000/TreeNode.swift

public class TreeNode {
    public var val: Int
    public var left: TreeNode?
    public var right: TreeNode?

    public init(_ val: Int) {
        self.val = val
        self.left = nil
        self.right = nil
    }

    class func getTreeNode(ints: Int...) -> TreeNode {
        var tns: [TreeNode] = []
        for var i = 0; i < ints.count; i++ {
            tns.append(TreeNode(ints[i]))
        }
        for var i = 0; i < tns.count; i++ {
            let left = 2 * i + 1, right = left + 1
            if (left >= tns.count) {break}
            tns[i].left = tns[left]
            if (right >= tns.count) {break}
            tns[i].right = tns[right]
        }
        return tns[0]
    }

    class func printTreeNode(tns: TreeNode...) {
        for tn in tns {
            print(String(p(tn)))
        }
    }

    class func p(tn: TreeNode?) -> [Character] {
        if tn == nil {return []}
        var bytes: [Character] = []
        bytes.append("[")
        bytes += p(tn!.left)
        bytes.append(Character(UnicodeScalar((tn!.val) + 48)))
        bytes += p(tn!.right)
        bytes.append("]")
        return bytes
    }
}

TreeNode.printTreeNode(TreeNode.getTreeNode(1, 5, 6, 2), TreeNode.getTreeNode(7, 8, 6, 2))
